src/main/java/uk/gov/hmcts/reform/sscs/messaging/HmcHearingsEventTopicListener.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.sscs.messaging;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusException;
+import com.azure.messaging.servicebus.ServiceBusFailureReason;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Data
+@SuppressWarnings({
+    "PMD.LawOfDemeter",
+    "PMD.CompareObjectsWithEquals",
+    "PMD.CloseResource"
+})
+public final class HmcHearingsEventTopicListener {
+
+    private final HmcHearingsTopicListenerProperties properties;
+
+    @SuppressWarnings("unused")
+    public void receiveMessages() {
+        CountDownLatch countdownLatch = new CountDownLatch(1);
+
+        ServiceBusProcessorClient processorClient = new ServiceBusClientBuilder()
+            .connectionString(properties.getConnectionString())
+            .processor()
+            .topicName(properties.getTopicName())
+            .subscriptionName(properties.getSubName())
+            .processMessage(HmcHearingsEventTopicListener::processMessage)
+            .processError(context -> processError(context, countdownLatch))
+            .buildProcessorClient();
+
+        processorClient.start();
+        try {
```

Comment:
I'm pretty sure that the Azure example added this sleep section purely for demo purposes. Remove the try block and the closing of the client, we want it listening for the lifetime of the application.

Reply:
ah ok @MohammedHMCTS did mention this, I thought we will listen to it on time oscillation. Will look into it.

Reply:
ah ok @MohammedHMCTS did mention this, I thought we will listen to it on time oscillation. Will look into it.